Mobile simulator fails to link to OpenGL

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Running `fyne_demo` after compiling with `-tags mobile` on develop results in the following crash:
```
panic: failed to link OpenGL program:


goroutine 16 [running]:
fyne.io/fyne/v2/internal/painter/gl.(*painter).createProgram(0xc0001122a0, {0xb207f3?, 0x9?})
        /home/jacob/git/fyne/internal/painter/gl/painter.go:174 +0x665
fyne.io/fyne/v2/internal/painter/gl.(*painter).Init(0xc0001122a0)
        /home/jacob/git/fyne/internal/painter/gl/gl_gomobile.go:73 +0x145
fyne.io/fyne/v2/internal/driver/mobile.(*mobileDriver).handlePaint(0xc0000de580, {0x40?}, {0xc550e8, 0xc000156000})
        /home/jacob/git/fyne/internal/driver/mobile/driver.go:250 +0x9a
fyne.io/fyne/v2/internal/driver/mobile.(*mobileDriver).Run.func1({0xc4f2c0?, 0x12eff40})
        /home/jacob/git/fyne/internal/driver/mobile/driver.go:188 +0x418
fyne.io/fyne/v2/internal/driver/mobile/app.main.func1()
        /home/jacob/git/fyne/internal/driver/mobile/app/x11.go:55 +0x34
created by fyne.io/fyne/v2/internal/driver/mobile/app.main
        /home/jacob/git/fyne/internal/driver/mobile/app/x11.go:54 +0x1ea
```

Master and release branch are both working fine.

### How to reproduce

1. Compile `fyne_demo` with `go build -tags mobile`.
2. Run `./fyne_demo`
3. Notice crash

### Screenshots

_No response_

### Example code

Any Fyne application will do.

### Fyne version

f0e5190448680df16bfda21f1554098a65af2cc3

### Go compiler version

Go 1.19.6

### Operating system and version

Fedora Linx 37

### Additional Information

_No response_

--- a/internal/driver/mobile/gl/consts.go
+++ b/internal/driver/mobile/gl/consts.go
@@ -49,6 +49,7 @@ const (
 
 	UnsignedByte = 0x1401
 	Float        = 0x1406
+	RED          = 0x1903
 	RGBA         = 0x1908
 	LUMINANCE    = 0x1909
 

--- a/internal/driver/mobile/gl/fn.go
+++ b/internal/driver/mobile/gl/fn.go
@@ -63,8 +63,8 @@ const (
 	glfnGetShaderiv
 	glfnGetTexParameteriv
 	glfnGetUniformLocation
-	glfnLinkProgram
 	glfnPixelStorei
+	glfnLinkProgram
 	glfnReadPixels
 	glfnScissor
 	glfnShaderSource

--- a/internal/driver/mobile/gl/gl.go
+++ b/internal/driver/mobile/gl/gl.go
@@ -387,7 +387,7 @@ func (ctx *context) PixelStorei(pname Enum, param int32) {
 		args: fnargs{
 			fn: glfnPixelStorei,
 			a0: pname.c(),
-			a2: uintptr(param),
+			a1: uintptr(param),
 		},
 	})
 }

--- a/internal/driver/mobile/gl/work_windows.go
+++ b/internal/driver/mobile/gl/work_windows.go
@@ -237,6 +237,10 @@ var glfnMap = map[glfn]func(c call) (ret uintptr){
 		ret, _, _ = syscall.Syscall(glGetUniformLocation.Addr(), 2, c.args.a0, c.args.a1, 0)
 		return ret
 	},
+	glfnPixelStorei: func(c call) (ret uintptr) {
+		syscall.Syscall(glPixelStorei.Addr(), 2, c.args.a0, c.args.a1, 0)
+		return
+	},
 	glfnLinkProgram: func(c call) (ret uintptr) {
 		syscall.Syscall(glLinkProgram.Addr(), 1, c.args.a0, 0, 0)
 		return
@@ -344,6 +348,7 @@ var (
 	glGetShaderiv             = libGLESv2.NewProc("glGetShaderiv")
 	glGetTexParameteriv       = libGLESv2.NewProc("glGetTexParameteriv")
 	glGetUniformLocation      = libGLESv2.NewProc("glGetUniformLocation")
+	glPixelStorei             = libGLESv2.NewProc("glPixelStorei")
 	glLinkProgram             = libGLESv2.NewProc("glLinkProgram")
 	glReadPixels              = libGLESv2.NewProc("glReadPixels")
 	glScissor                 = libGLESv2.NewProc("glScissor")

--- a/internal/painter/gl/gl_const_darwin.go
+++ b/internal/painter/gl/gl_const_darwin.go
@@ -1,0 +1,9 @@
+package gl
+
+import (
+	"fyne.io/fyne/v2/internal/driver/mobile/gl"
+)
+
+const (
+	singleChannelColorFormat = gl.RED
+)

--- a/internal/painter/gl/gl_const_other.go
+++ b/internal/painter/gl/gl_const_other.go
@@ -10,3 +10,5 @@ import (
 const (
 	singleChannelColorFormat = gl.LUMINANCE
 )
+
+var _ = singleChannelColorFormat

--- a/internal/painter/gl/gl_const_other.go
+++ b/internal/painter/gl/gl_const_other.go
@@ -1,5 +1,5 @@
-//go:build !darwin
-// +build !darwin
+//go:build !darwin && !js && !wasm
+// +build !darwin,!js,!wasm
 
 package gl
 

--- a/internal/painter/gl/gl_const_other.go
+++ b/internal/painter/gl/gl_const_other.go
@@ -1,0 +1,12 @@
+//go:build !darwin
+// +build !darwin
+
+package gl
+
+import (
+	"fyne.io/fyne/v2/internal/driver/mobile/gl"
+)
+
+const (
+	singleChannelColorFormat = gl.LUMINANCE
+)

--- a/internal/painter/gl/gl_gomobile.go
+++ b/internal/painter/gl/gl_gomobile.go
@@ -18,7 +18,7 @@ const (
 	bitDepthBuffer        = gl.DepthBufferBit
 	clampToEdge           = gl.ClampToEdge
 	colorFormatRGBA       = gl.RGBA
-	colorFormatR          = gl.LUMINANCE
+	colorFormatR          = singleChannelColorFormat
 	compileStatus         = gl.CompileStatus
 	constantAlpha         = gl.ConstantAlpha
 	float                 = gl.Float

